File: 02_ProgrammingFundamentals/_09_MapsLambdaAndStreamAPI/_02_Exercise/_04_SoftUniParking.java
package _02_ProgrammingFundamentals._09_MapsLambdaAndStreamAPI._02_Exercise;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Scanner;

public class _04_SoftUniParking {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        int numberOfCommands = Integer.parseInt(scan.nextLine());

        // Add variable
        Map<String, String> registry = new LinkedHashMap<>();

        // Validate parking permissions and save data
        for (int i = 0; i < numberOfCommands; i++) {
            String[] commands = scan.nextLine().split("\\s+");

            String command = commands[0];
            String username = commands[1];

            if (command.equals("register")) {
                String licensePlate = commands[2];
                
                if (registry.containsKey(username)) {
                    System.out.printf("ERROR: already registered with plate number %s%n", licensePlate);
                    continue;
                }
                
                registry.put(username, licensePlate);
                System.out.printf("%s registered %s successfully%n", username, licensePlate);
            } else {
                if (!registry.containsKey(username)) {
                    System.out.printf("ERROR: user %s not found%n", username);
                    continue;
                }
                
                registry.remove(username);
                System.out.printf("%s unregistered successfully%n", username);
            }
        }

        // Print result
        registry.forEach((key, value) -> System.out.printf("%s => %s%n", key, value));

        scan.close();
    }
}

File: 24_C#Fundamentals/_07_AssociativeArrays/_02_Exercise/_04_SoftUniParking/_04_SoftUniParking.cs
using System;

public class _04_SoftUniParking
{
    public static void Main(string[] args)
    {
        // Read input value
        int numberOfCommands = int.Parse(Console.ReadLine());

        // Add variable
        Dictionary<string, string> registry = new Dictionary<string, string>();

        // Validate parking permissions and save data
        for (int i = 0; i < numberOfCommands; i++)
        {
            string[] commands = Console.ReadLine().Split(" ");

            string command = commands[0];
            string username = commands[1];

            if (command.Equals("register"))
            {
                string licensePlate = commands[2];

                if (registry.ContainsKey(username))
                {
                    Console.WriteLine($"ERROR: already registered with plate number {licensePlate}");
                    continue;
                }

                registry[username] = licensePlate;
                Console.WriteLine($"{username} registered {licensePlate} successfully");
            }
            else
            {
                if (!registry.ContainsKey(username))
                {
                    Console.WriteLine($"ERROR: user {username} not found");
                    continue;
                }

                registry.Remove(username);
                Console.WriteLine($"{username} unregistered successfully");
            }
        }

        // Print result
        foreach (KeyValuePair<string, string> reg in registry) 
        {
            Console.WriteLine($"{reg.Key} => {reg.Value}");
        }
    }
}
